### PR TITLE
Connect with home office PRP1

### PR DESCRIPTION
**Pull request by divisathyan**
Connect with home office PRP1

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3509

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java [hunk 1]
```diff
@@ -31,8 +34,19 @@ public HomeOfficeSearchService(
     public HomeOfficeSearchResponse getCaseStatus(String homeOfficeReferenceNumber) {
 
         HomeOfficeSearch request = makeRequestBody(homeOfficeReferenceNumber);
+        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+        try {
+            log.info("HomeOffice-CaseSearch request: {}", ow.writeValueAsString(request));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java:41]: shouldn't eat up the Exception here.  Just re-throw it so we can see it in logs and in appinsights.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java [hunk 2]
```diff
@@ -31,8 +34,19 @@ public HomeOfficeSearchService(
...
+        }
+
         HomeOfficeSearchResponse searchResponse = homeOfficeSearchApi.getStatus(request);
-        log.info("HomeOffice-CaseSearch response: {}", searchResponse);
+        try {
+            log.info("HomeOffice-CaseSearch response: {}", ow.writeValueAsString(searchResponse));
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
```
Comment (raghera) [inline src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java:48]: see comment above.

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java (hunk 1 above)
Comment (raghera) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java:41]: Not sure it needs a try/catch block?

Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java:41]: Throw exception and removed exception handling

File: src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java (hunk 2 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/service/HomeOfficeSearchService.java:48]: same fix as above comment

**Pull request merged**